Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -454,12 +454,11 @@ internal async Task<GoogleCredential> GetGoogleCredentialAsync(CancellationToken
         protected abstract ChannelPool GetChannelPool();
 
         /// <summary>
-        /// Returns the default <see cref="GrpcAdapter"/> to use if
-        /// <see cref="GrpcAdapter"/> is not set.
+        /// 
         /// </summary>
-        protected abstract GrpcAdapter DefaultGrpcAdapter { get; }
+        protected abstract GrpcApiDescriptor GrpcApiDescriptor { get; }
 
-        private GrpcAdapter EffectiveGrpcAdapter => GrpcAdapter ?? DefaultGrpcAdapter;
+        private GrpcAdapter EffectiveGrpcAdapter => GrpcAdapter ?? GrpcAdapter.GetFallbackAdapter(GrpcApiDescriptor);
```

Comment:
This is not taking DI into acount right now, right? Which is fine as that hasn't been merged, just answered myself.

Reply:
Correct. Although the DI part would set `GrpcAdapter` anyway, so it might not make much difference.